### PR TITLE
Moving journal-local-demo into its own folder

### PR DESCRIPTION
**Pull request by giorgiosironi**
Deploys and enables the virtual host, but the parameters need to be tweaked to disable Redis and similar dependencies.

### CODE REVIEW

**Comment by giorgiosironi** [comment]
Ran into some problems with the `nginx-server-service` state, which `watch`es all sites in `sites-enabled`. This causes mutual dependencies because operations that touch the virtual hosts there are pushed up in the highstate. Either we do this with `listen` instead, if possible, or we do it manually when deploying the single virtual hosts.

**Comment by thewilkybarkid** [comment]
@giorgiosironi Good to merge? (builder-private will need an update due to b1f6fad0d4b7273c18d505d1a3f4e770edfa5e98.)

**Comment by thewilkybarkid** [comment]
Just spotted an rsync problem.

File: salt/journal/critical-css.sls [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        - require:
+            - journal-local-demo-parameters
+
+journal-local-demo-nginx-vhost:
+    file.managed:
+        - name: /etc/nginx/sites-enabled/journal-local-demo.conf
+        - source: salt://journal/config/etc-nginx-sites-enabled-journal-local-demo.conf
+        - template: jinja
+        - require:
+            - nginx-config
```
Comment (giorgiosironi) [inline salt/journal/critical-css.sls]: this has no `require` on `journal-local-demo-parameters`, which could trigger the recursion problem. Should be ok

Comment (thewilkybarkid) [inline salt/journal/critical-css.sls]: `generate-critical-css` does (through `journal-local-demo-cache-clean`).

**Comment by giorgiosironi** [comment]
Doing the last provisioning from scratch...

**Comment by giorgiosironi** [comment]
:+1: ready to merge

**Pull request merged**